test/functional/options/winfixbuf_spec.lua

```diff
@@ -45,11 +45,31 @@ describe("Nvim API calls with 'winfixbuf'", function()
       end
 
       local other_buffer = _setup_two_buffers()
-      local results, _ = pcall(vim.api.nvim_set_current_buf, other_buffer)
+      local ok, _ = pcall(vim.api.nvim_set_current_buf, other_buffer)
 
-      return results
+      return ok
     ]])
 
-    assert(results == false)
+    assert(not ok)
+  end)
+
+  it('vim.api.nvim_win_set_buf on current buffer', function()
+    local ok = exec_lua([[
+      vim.wo.winfixbuf = true
+      local ok, _ = pcall(vim.api.nvim_win_set_buf, 0, vim.api.nvim_get_current_buf())
+      return ok
+    ]])
+
+    assert(ok)
```

Comment:
sorry, just noticed that `pcall` is not needed here because we don't expect failure. same for the other case. in order to actually assert something useful, maybe we should check that the buffer didn't change.

```suggestion
      local curbuf = vim.api.nvim_get_current_buf()
      vim.api.nvim_win_set_buf(0, curbuf)
      assert(vim.api.nvim_get_current_buf() == curbuf)
    ]])
```